src/components/checkbox/checkbox.scss

```diff
@@ -125,21 +121,22 @@
     &:disabled {
       background-color: transparent; /* 1 */
       border-style: dashed; /* 1 */
-      color: colors.$gray-dark; /* 2 */
+      color: var(--theme-color-text-checkbox-disabled); /* 2 */
       cursor: not-allowed;
 
-      /**
-       * For purely visual consistency with dark theme disabled inputs we would
-       * use `$primary-brand-lighter`, but that will not have enough contrast
-       * on its own in this context.
-       */
-
-      .t-dark & {
-        color: colors.$text-light;
-      }
-
       &::after {
-        background-image: svg-load('icons/check.svg', stroke=colors.$gray-dark);
+        /**
+         * Inline SVGs aren't aware of custom properties, so we have to do a bit
```

Comment:
Nit: I was a little confused by this comment since I thought you meant inlining the SVG code in HTML, not using an SVG as a background image.